Modules/_io/bytesio.c

```diff
@@ -609,8 +609,9 @@ _io_BytesIO_readinto_impl(bytesio *self, Py_buffer *buffer)
     n = self->string_size - self->pos;
     if (len > n) {
         len = n;
-        if (len < 0)
-            len = 0;
+        if (len < 0) {
+            return PyLong_FromSsize_t(0);
+        }
     }
 
     assert(self->pos + len < PY_SSIZE_T_MAX);
```

Comment:
Isn't it enough to fix this assertion?
```suggestion
    assert(self->pos + len <= PY_SSIZE_T_MAX);
```

Reply:
It will but I worry about the later `PyBytes_AS_STRING(self->buf) + self->pos` as `self->pos` is `PY_SSIZE_T_MAX` and adding a large integer to a pointer then doing a `memcpy` from that computed address (even 0 bytes) feels like a path to undefined behavior to me. Happy to implement whichever you'd prefer.

Reply:
Good point. If adding PY_SSIZE_T_MAX causes troubles, adding slightly smaller value will also cause troubles. Actually, if pos larger than the size of the array, ot is an undefined behaviour. So, a specoal case for empty read is needed too.

Reply:
Please update also the assert.